mess/workflow/rules/processing/reads.smk

```diff
@@ -537,6 +540,146 @@ if not SKIP_SHUFFLE:
             paste -d '\t' <(seqkit seq -n {output}) <(seqkit seq -n {input}) > {log[1]} 
             """
 
+if ERRFREE:
+    rule fix_art_sam_ef:
+        """
+        rule to replace SAM cigar string with read length + M
+        Fixes truncated art_illumina SAM files with some genomes
+        """
+        input:
+            os.path.join(fastq_dir, "{sample}", "{fasta}", contig + "_ef.sam"),
+        output:
+            temp(os.path.join(fastq_dir, "{sample}", "{fasta}", contig + "_ef.fixed")),
+        resources:
+            mem_mb=config.resources.sml.mem,
+            mem=str(config.resources.sml.mem) + "MB",
+            time=config.resources.sml.time,
+        params:
+            maxlen=MEAN_LEN,
+        shell:
+            """
+            awk 'BEGIN {{OFS="\t"}} {{ if ($1 ~ /^@/) {{ print $0 }} \\
+            else {{ $6 = "{params.maxlen}M"; print $0 }} }}' \\
+            {input} > {output}
+            """
+    
+    rule convert_sam_to_bam_ef:
```

Comment:
I simplified the rules below in the [readproc branch](https://github.com/metagenlab/MeSS/tree/readproc), It would be good to adapt the PR according to that branch !